src/neo/Network/P2P/Payloads/Witness.cs

```diff
@@ -48,5 +49,18 @@ public JObject ToJson()
             json["verification"] = Convert.ToBase64String(VerificationScript);
             return json;
         }
+
+        private WitnessFlag _flag = WitnessFlag.None;
+        public WitnessFlag Flag
+        {
+            get
+            {
+                if (_flag == WitnessFlag.None)
+                {
+                    _flag = VerificationScript.Length == 0 ? WitnessFlag.StateDependentWitness : WitnessFlag.StateIndependentWitness;
```

Comment:
So all deployed scripts are state dependent and the others are state independent?

Reply:
Yes, here the filter logic is the same as SmartContract.Helper.VerifyWitnesses